include/rdm/param_data.h

```diff
@@ -0,0 +1,585 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************
+ * This file is a part of RDM. For more information, go to:
+ * https://github.com/ETCLabs/RDM
+ ******************************************************************************/
+
+/*! \file rdm/param_data.h
+ *  \brief Structures and packing/unpacking functions for the parameter data of standard PIDs.
+ *  \author Christian Reese*/
+#ifndef _RDM_PARAM_DATA_H_
+#define _RDM_PARAM_DATA_H_
+
+#include "etcpal/bool.h"
+#include "etcpal/error.h"
+#include "etcpal/inet.h"
+#include "rdm/message.h"
+
+/*! The maximum length of a typical string in the parameter data, including one extra byte for a null terminator. */
+#define RDMPD_STRING_MAX_LENGTH 33
+
+/*! A structure representing a typical string in the parameter data. */
+typedef struct RdmPdString
+{
+  char string[RDMPD_STRING_MAX_LENGTH]; /*!< The string data. */
+} RdmPdString;
+
+/* e1.33 Table 7-1: Required E1.20 Parameter IDs for the Default Responder */
+// Get/Set IDENTIFY_DEVICE See Section 10.11.1 of [RDM]
+etcpal_error_t rdmpd_unpack_get_resp_identify_device(const RdmParamData *pd, bool *identify_state_out);
+etcpal_error_t rdmpd_pack_get_resp_identify_device(bool identify_state, RdmParamData *pd_out);
+etcpal_error_t rdmpd_unpack_set_identify_device(const RdmParamData *pd, bool *identify_out);
+etcpal_error_t rdmpd_pack_set_identify_device(bool identify, RdmParamData *pd_out);
+
+// Get SUPPORTED_PARAMETERS See Section 10.4.1 of [RDM]
+/*! The maximum number of 16-bit parameter IDs in the Get Supported Parameters Response PD. */
+#define RDMPD_MAX_SUPPORTED_PARAMETERS 115
+
+/*! A structure representing supported parameters. */
+typedef struct RdmPdSupportedParameters
+{
+  uint16_t parameters[RDMPD_MAX_SUPPORTED_PARAMETERS]; /*!< The array of supported PIDs. */
+  size_t length;                                       /*!< The number of supported PIDs in the array. */
+} RdmPdSupportedParameters;
+
+etcpal_error_t rdmpd_unpack_get_resp_supported_parameters(const RdmParamData *pd, RdmPdSupportedParameters *params_out);
+etcpal_error_t rdmpd_pack_get_resp_supported_parameters(const RdmPdSupportedParameters *params, RdmParamData *pd_out);
+
+// Get PARAMETER_DESCRIPTION See Section 10.4.2 of [RDM]
+etcpal_error_t rdmpd_unpack_get_parameter_description(const RdmParamData *pd, uint16_t *requested_pid_out);
+etcpal_error_t rdmpd_pack_get_parameter_description(uint16_t requested_pid, RdmParamData *pd_out);
+
+/*! Data type enumeration */
+typedef enum
```

Comment:
These enumerations are right out of the RDM spec, right?  I believe we have a header for all of these values.  I'm not sure we need them as actual enums, or if we do they should be defined to = the constants.

Reply:
I like having them as enums, for the small bit of added type safety. I think we should:
```
#include rdm/defs.h
// ...
  kRdmPdDtNotDefined = E120_DS_NOT_DEFINED
  // etc...
```

Reply:
I added the constants to MOST of the enums, besides rdmpd_broker_state_t (because RDMnet consts are not available to the RDM library) and rdmpd_backgnd_qd_status_policy_setting_t (because the values e1.37-7 specifies are different from what are defined in the RDM library).

Reply:
After an offline discussion with Sam, rdmpd_broker_state_t now has constants because those constants were moved from the RDMnet library to the RDM library. As for rdmpd_backgnd_qd_status_policy_setting_t, we are going to leave it as it is for now.